### PR TITLE
test: switch to titanoboa

### PR DESCRIPTION
**Pull request by tserg**
### What I did

Use titanoboa for test suite.

### How I did it

Change fixtures to boa.

### How to verify it

See updated tests.

### Commit message

WIP

### Description for the changelog

WIP

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: tests/base_conftest.py [hunk 1]
```diff
@@ -215,8 +215,8 @@ def get_logs(tx_hash, c, event_name):
 
 
 @pytest.fixture(scope="module")
-def assert_w3_tx_failed(tester):
-    def assert_w3_tx_failed(function_to_test, exception=TransactionFailed, exc_text=None):
+def w3_assert_tx_failed(tester):
+    def w3_assert_tx_failed(function_to_test, exception=TransactionFailed, exc_text=None):
```
Comment (fubuloubu) [inline tests/base_conftest.py:220]: Why rename all of this?

Comment (tserg) [reply tests/base_conftest.py:220]: There are two sets of tests with this change that require different fixtures to set up the test cases - those that will use titanoboa, and those that will use w3 due to certain features such as block mining. Therefore, I renamed the existing w3 fixtures with a `w3` prefix, and switched to titanoboa for the existing fixture names.

Comment (charles-cooper) [reply tests/base_conftest.py:220]: hmm. i think we can get rid of `assert_tx_failed` entirely and replace with uses of `boa.reverts()`. are there test cases which really depend on block mining / sealing?

**Comment by charles-cooper** [comment]
one issue that i'm realizing is an issue is that (via API changes) vyper could break titanoboa in the future, and because of how the dependencies are set up, we will have a chicken/egg problem with updating the dependencies.

**Comment by tserg** [comment]
> one issue that i'm realizing is an issue is that (via API changes) vyper could break titanoboa in the future, and because of how the dependencies are set up, we will have a chicken/egg problem with updating the dependencies.

Good point, I agree. I think this alone is reason enough to close this PR.

It also feels circular to be testing vyper with titanoboa that depends on vyper's API itself. I now think that switching to titanoboa would be less robust.

**Pull request closed without merging**